x-pack/plugins/security/public/management/roles/edit_role/edit_role_page.tsx

```diff
@@ -452,45 +453,68 @@ export const EditRolePage: FunctionComponent<Props> = ({
-            <FormattedMessage
-              id="xpack.security.management.editRole.roleNameFormRowTitle"
-              defaultMessage="Role name"
-            />
-          }
-          helpText={
-            !isEditingExistingRole ? (
-              <FormattedMessage
-                id="xpack.security.management.createRole.roleNameFormRowHelpText"
-                defaultMessage="Once the role is created you can no longer edit its name."
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <EuiFormRow
+              data-test-subj={'roleNameFormRow'}
+              label={
+                <FormattedMessage
+                  id="xpack.security.management.editRole.roleNameFormRowTitle"
+                  defaultMessage="Role name"
+                />
+              }
+              helpText={
+                !isEditingExistingRole ? (
+                  <FormattedMessage
+                    id="xpack.security.management.createRole.roleNameFormRowHelpText"
+                    defaultMessage="Once the role is created you can no longer edit its name."
+                  />
+                ) : !isRoleReserved ? (
+                  <FormattedMessage
+                    id="xpack.security.management.editRole.roleNameFormRowHelpText"
+                    defaultMessage="A role's name cannot be changed once it has been created."
+                  />
+                ) : undefined
+              }
+              {...validator.validateRoleName(role)}
+              {...(creatingRoleAlreadyExists
+                ? { error: 'A role with this name already exists.', isInvalid: true }
+                : {})}
+            >
+              <EuiFieldText
+                name={'name'}
+                value={role.name || ''}
+                onChange={onNameChange}
+                onBlur={onNameBlur}
+                data-test-subj={'roleFormNameInput'}
+                disabled={isRoleReserved || isEditingExistingRole || isRoleReadOnly}
+                isInvalid={creatingRoleAlreadyExists}
               />
-            ) : !isRoleReserved ? (
-              <FormattedMessage
-                id="xpack.security.management.editRole.roleNameFormRowHelpText"
-                defaultMessage="A role's name cannot be changed once it has been created."
+            </EuiFormRow>
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiFormRow
+              data-test-subj="roleDescriptionFormRow"
+              label={
+                <FormattedMessage
+                  id="xpack.security.management.editRole.roleDescriptionFormRowTitle"
+                  defaultMessage="Role description"
+                />
+              }
+            >
+              <EuiFieldText
+                name="description"
+                value={role.description ?? ''}
+                onChange={onDescriptionChange}
+                data-test-subj="roleFormDescriptionInput"
+                disabled={isRoleReserved || isRoleReadOnly}
```

Comment:
Nit: If possible, can we use built-in truncation to show ellipses for long strings? And in the case of the read-only view also including a tooltip to show the full description like we do in the roles combo box?

<img width="428" alt="Screenshot 2024-05-14 at 6 59 47 PM" src="https://github.com/elastic/kibana/assets/103939324/9115744b-2388-4e33-8615-39e6f92d21c9">

Reply:
`EuiFieldText` [doesn't support truncation and tooltip](https://eui.elastic.co/#/forms/form-controls#text-field), it means we would need to change it to another component. Instead of rendering it we can show `TruncatedText` component, but in this case we won't match the design think

Reply:
Added tooltip to the input in the read-only view.